snuba/subscriptions/codecs.py

```diff
@@ -1,41 +1,23 @@
 import json
-from datetime import timedelta
 
 from snuba.query.exceptions import InvalidQueryException
-from snuba.subscriptions.data import SubscriptionData
+from snuba.subscriptions.data import LegacySubscriptionData, SubscriptionData
 from snuba.subscriptions.worker import SubscriptionTaskResult
 from snuba.utils.codecs import Codec, Encoder
 from snuba.utils.streams.backends.kafka import KafkaPayload
 
 
 class SubscriptionDataCodec(Codec[bytes, SubscriptionData]):
     def encode(self, value: SubscriptionData) -> bytes:
```

Comment:
I am not 100% sure whether to have one codec that generates instances of both legacy and snql or whether having two separate codecs (and thus two separate storages). 
The advantage of the second is that we have cleaner and simpler codec that respect the original intention. 
The disadvantage is that then we need to have two separate paths all the way down from views.py (and in the consumer as well). 

Will figure this out in the next PR in this series, the change here would be marginal in either case. The difference would be that. In case of option 2 `to_dict` and `from_dict` would not be necessary anymore.